src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,50 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(health = 100, name) {
+    this.health = health;
+    this.name = name;
+    Animal.alive.push(this); // Add the instance to the alive array when created
+  }
 }
+Animal.alive = [];
 
-class Herbivore extends Animal {
-  // write your code here
+class Carnivore extends Animal {
+  constructor(name) {
+    super(100, name);
+  }
+
+  bite(target) {
+    if (target instanceof Herbivore && !target.hidden) {
+      const damage = Math.min(target.health, 50);
+
+      target.health -= damage;
+      target.checkIfDead();
+    }
+  }
 }
 
-class Carnivore extends Animal {
-  // write your code here
+class Herbivore extends Animal {
+  constructor(name, hidden = false) {
+    super(100, name);
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
```

Comment:
```suggestion
    this.hidden = true;
```

Hide method should not toggle hidden value and only change it to true